### PR TITLE
Add Canary Testing Documentation

### PR DESCRIPTION
**Pull request by victorhuangwq**
- Updates `README.md` with instructions for testing new WebView2 runtime changes using Edge Canary.
- Adds `test_canary.md` detailing the process to test features in Edge Canary, including download steps and registry key setup.

Aims to streamline diagnostics collection and facilitate early testing of WebView2 runtime updates for developers

### CODE REVIEW

File: diagnostics/test_canary.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Test Feature in Canary
+
+The latest runtime changes are first shipped in the WebView2 runtime in the Edge Canary Browser. If you would like to test the new changes with your application, here's a suggested way to do so:
```
Comment (vbryh-msft) [inline diagnostics/test_canary.md:3]: nit: "to test the new changes " -> + or help us to verify that your issue is fixed

File: diagnostics/test_canary.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+1. Download [Edge Canary](https://www.microsoft.com/en-us/edge/download/insider)
+2. Run the following command in a PowerShell with Administrative Priviledges:  
+  `REG ADD HKCU\Software\Policies\Microsoft\Edge\WebView2\ChannelSearchKind /v <your_app.exe> /t REG_DWORD /d 1`
+3. Restart your application, and you should be able to see that your application picks up the latest runtime.
```
Comment (vbryh-msft) [inline diagnostics/test_canary.md:8]: Do you know if it is ok to restart only app - or they need to restart shell also like with env vars?

File: diagnostics/test_canary.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+1. Download [Edge Canary](https://www.microsoft.com/en-us/edge/download/insider)
+2. Run the following command in a PowerShell with Administrative Priviledges:  
+  `REG ADD HKCU\Software\Policies\Microsoft\Edge\WebView2\ChannelSearchKind /v <your_app.exe> /t REG_DWORD /d 1`
+3. Restart your application, and you should be able to see that your application picks up the latest runtime. 
+
+To remove the registry key added, run the following:  
+`REG DELETE HKCU\Software\Policies\Microsoft\Edge\WebView2\ChannelSearchKind`
+
+For more information, read the documentation on [Test upcoming APIs and features](https://learn.microsoft.com/en-us/microsoft-edge/webview2/how-to/set-preview-channe)
```
Comment (david-risney) [inline diagnostics/test_canary.md:13]: The URI doesn't resolve - missing l at the end

File: diagnostics/test_canary.md (hunk 2 above)
Comment (victorhuangwq) [reply diagnostics/test_canary.md:8]: @vickiez they only need to restart the application once the reg key is applied, right?

File: diagnostics/test_canary.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+The latest runtime changes are first shipped in the WebView2 runtime in the Edge Canary Browser. If you would like to test the new changes or help us to verify that your issue is fixed with your application, here's a suggested way to do so:
+
+1. Download [Edge Canary](https://www.microsoft.com/en-us/edge/download/insider)
+2. Run the following command in a PowerShell with Administrative Priviledges:  
+  `REG ADD HKCU\Software\Policies\Microsoft\Edge\WebView2\ChannelSearchKind /v <your_app.exe> /t REG_DWORD /d 1`
+3. Restart your application, and you should be able to see that your application picks up the latest runtime. 
+
+To remove the registry key added, run the following:  
+`REG DELETE HKCU\Software\Policies\Microsoft\Edge\WebView2\ChannelSearchKind`
```
Comment (vbryh-msft) [inline diagnostics/test_canary.md:11]: the command line for adding regkey is for particular app, but removing one is for all?

Comment (victorhuangwq) [reply diagnostics/test_canary.md:11]: hmm I was trying to make it simple. Let me edit it

Comment (vbryh-msft) [reply diagnostics/test_canary.md:11]: can do both - or mention that it will remove others. I agree about making it simple

Comment (victorhuangwq) [reply diagnostics/test_canary.md]: I removed it

**Review by vbryh-msft — APPROVED** [review]

**Pull request merged**